backend/lib/src/data/storage/memory.rs

```diff
@@ -44,22 +109,134 @@ impl Storage for InMemoryStorage {
     type Error = InMemoryStorageError;
 
     async fn health_check(&self) -> Result<bool, Self::Error> {
-        // just to "use" `map`
-        // TODO(SCAFFOLDING): implement a proper check for the storage once we defined what it needs to contain
-        #[allow(unused_comparisons)]
-        #[allow(clippy::absurd_extreme_comparisons)]
-        Ok(self.map.read().capacity() >= 0)
+        Ok(true)
+    }
+
+    async fn store_nonce(
+        &self,
+        message: String,
+        address: String,
+        expiration_seconds: u64,
+    ) -> Result<(), Self::Error> {
+        let now = Instant::now();
+        let expires_at = now + Duration::from_secs(expiration_seconds);
+
+        let entry = NonceEntry {
+            address,
+            expires_at,
+        };
+
+        self.nonces.write().insert(message, entry);
+        Ok(())
+    }
+
+    async fn get_nonce(&self, message: &str) -> Result<Option<String>, Self::Error> {
```

Comment:
will this ever return Error? Is it necessary to wrap the return value in a Result? 🤔

Reply:
Also, if the storage ever becomes something different like redis, we might get network errors